### PR TITLE
[Feature] Display student code in new component and fix position of lock repo label

### PR DESCRIPTION
**Pull request by fde312**
### Checklist
- [x] I tested *all* changes and *all* related features with different users (student, tutor, instructor, admin) **locally**
- [x] Client: I documented the TypeScript code using JSDoc style.
- [x] Client: I added multiple screenshots/screencasts of my UI changes

### Motivation and Context

We introduce a new component which displays the student code of programming exercises without using ace editor. This component is currently not in use but will then be used for the manual assessment.

### Description

- Introduce new component to display student code
- Fixed position of lock repo label

### Steps for Testing

**NOTE: You can test this only locally with the programming setup at the moment**
To test it you need to adapt the following first:
- Replace `<jhi-code-editor-ace>` component in `code-editor-container.html` with following:
`<jhi-code-editor-tutor-assessment-display-code editorCenter [selectedFile]="selectedFile" (onError)="onError($event)"></jhi-code-editor-tutor-assessment-display-code>`
- Import the new component to `code-editor.module`or you can press `alt + enter`this should do it for you 😊 

1. Log in to Artemis
2. Navigate to Course Administration
3. Open up the code editor
4. Make sure that the new component is loaded and displays the file data correctly

Syntax highlighting will be added in the future.

### Screenshots

image

video1

Position of lock repo label now:
image

### CODE REVIEW

**Review by kloessst — APPROVED** [review]
Tested this locally and it works fine for me. Code looks good to me too!

**Review by sleiss — APPROVED** [review]
Tested it locally, works for me.

File: src/main/webapp/app/exercises/programming/assess/code-editor-tutor-assessment-display-code.component.ts [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+        }
+    }
+
+    /**
+     * Setup the component after a file change occurred.
+     **/
+    initEditorAfterFileChange() {
+        if (this.selectedFile && this.fileSession[this.selectedFile]) {
+            this.studentCode = this.fileSession[this.selectedFile].code;
+            this.studentCodePerLine = this.studentCode.split('\n');
```
Comment (krusche) [inline src/main/webapp/app/exercises/programming/assess/code-editor-tutor-assessment-display-code.component.ts:48]: does this also work, when students use Windows line endings?
Can you double check how other code editors split lines?
This approach looks to simple to me and might be error prone, maybe there are even helper method that can do this in a better way

Comment (krusche) [reply src/main/webapp/app/exercises/programming/assess/code-editor-tutor-assessment-display-code.component.ts:48]: also see e.g. https://stackoverflow.com/questions/5034781/js-regex-to-split-by-line
(however keep in mind that we do not want to remove empty lines, nor do we want to remove whitespace)

Comment (fde312) [reply src/main/webapp/app/exercises/programming/assess/code-editor-tutor-assessment-display-code.component.ts]: I am using now `this.studentCodePerLine = this.studentCode.split(/\r?\n/);` too catch the windows endings. Alternatively we could set the ace editor to only allow unix line breaks with `editor.session.setNewlineMode('unix')`

File: src/main/webapp/app/exercises/programming/assess/code-editor-tutor-assessment-display-code.component.html [hunk 2]
```diff
@@ -0,0 +1,25 @@
+<div class="card">
+    <div class="card-header bg-primary text-white">
+        <h3 class="card-title">
+            <fa-icon [icon]="['far', 'file-alt']"></fa-icon>
+            <span> {{ selectedFile }}</span>
```
Comment (fde312) [inline src/main/webapp/app/exercises/programming/assess/code-editor-tutor-assessment-display-code.component.html:5]: test

**Review by kloessst — APPROVED** [review]
Re-approve

**Review by sleiss — APPROVED** [review]

**Pull request closed without merging**